src/main/java/com/arpnetworking/metrics/mad/PeriodWorker.java

```diff
@@ -305,6 +305,10 @@ private void processRecord(final Record record) {
     private final Bucket.Builder _bucketBuilder;
     private final NavigableMap<ZonedDateTime, Bucket> _bucketsByStart = new TreeMap<>();
     private final NavigableMap<ZonedDateTime, List<Bucket>> _bucketsByExpiration = new TreeMap<>();
+
+    // WARNING: Consider carefully the volume of samples recorded.
+    // PeriodicMetrics reduces the number of scopes creates, but each sample is
+    // still stored in-memory until it is flushed.
```

Comment:
You may want to add this to the docstring on the metrics client itself, since this advice still goes outside of this project.

Reply:
The periodic metrics class is experimental at the moment. This has been one of the learnings I've been discussing with @BrandonArp and as we fold it into the core library it will be addressed. However, in the meantime we are using the experimental functionality in MAD and CAGG to facilitate instrumentation.